### PR TITLE
Add logo section both fronted and backed

### LINKED ISSUE #354: Add option to "Add logo"
1. In the beginning of the form add a section "Event Name", where the user can enter the name of the event as text.

2. After the section "Choose your Badge Background" add a section "Use Logo". Implement as follows:
* add radio buttons a) "Add event logo" b) Use event name text
* If user chooses "Add event logo" show an upload button for the image file

### PR DESCRIPTION
**Pull request by YogeshSharma0201**
Fixes #354

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Added a section in create badges form on fronted for logo information
screenshot from 2018-12-30 13-09-05

- Changed cropper so that both logo and badge background can be cropped
screenshot from 2018-12-30 13-10-44

- changed preview section so that logo area can be filled separately by image or text
screenshot from 2018-12-30 13-12-17

- Necessary changes to backend are made and all the svgs files are modified
screenshot from 2018-12-30 13-14-36

badge

### CODE REVIEW

File: frontend/app/templates/components/user-component/preview.hbs [hunk 1]
```diff
@@ -1,42 +1,59 @@
...
-            style="
-            font-size : {{rel font_sh}}px;
-            font-family : {{defFont}};">
-            {{sh}}
-            </p>
+    <div class="bgImg ui raised segment" style="padding:0">
+        <div
+            style="background-size: cover;
+            background-repeat: no-repeat;
+            height: {{if previewHeight '105' '120'}}px;
```
[the remark this answers is not in the document]
Comment (YogeshSharma0201) [reply frontend/app/templates/components/user-component/preview.hbs]: logo height + badge body height = 475 or 600

File: api/controllers/generateBadges.py [hunk 2]
```diff
@@ -45,6 +45,9 @@ def generateBadges():
     csv_name = data.get('csv')
     badge_name = data.get('badge_name') or 'My Badge'
     image_name = data.get('image')
+    logo_image = data.get('logo_image')
+    logo_text = data.get('logo_text') or ''
```
[the remark this answers is not in the document]
Comment (YogeshSharma0201) [reply api/controllers/generateBadges.py]: On fronted by default it says fossaisa, on backend this is the desired functionality, if no logo_text print "" on badges

**Comment by YogeshSharma0201** [comment]
@rajvaibhavdubey isn't it already up-to-date?

File: frontend/app/controllers/create-badges.js [hunk 3]
```diff
@@ -29,30 +29,36 @@ export default Controller.extend({
...
+  logoImgFile    : '',
   badgeSize      : '',
   previewToggled : true,
   previewHeight  : '',
   badgeGenerated : false,
   backLink       : APP.backLink,
   defPaperSize   : '',
   genBadge       : '',
-  defImageName   : '',
+  defImageName   : 'black_futuristic',
```
[the remark this answers is not in the document]
Comment (YogeshSharma0201) [reply frontend/app/controllers/create-badges.js]: changed that

**Comment by gabru-md** [comment]
@sk9331657 is this PR good to go?

**Comment by sk9331657** [comment]
@YogeshSharma0201  Can you please provide a working gif of your pr or you can deploy your branch oh heroku?? You just need to connect your branch to Heroku app and provide  the link in the comment section here.

**Comment by YogeshSharma0201** [comment]
frontend link: https://arcane-castle-67878.herokuapp.com/create-badges
backend link: https://badgeyayapi.herokuapp.com
They both are connected.
@rajvaibhavdubey @sk9331657

**Comment by sk9331657** [comment]
@YogeshSharma0201 Work looks good. Try to keep max length of Event Name text restricted such that it doesn't overflow and gets out of badge. Long Strings should be restricted.

**Comment by rajvaibhavdubey** [comment]
@YogeshSharma0201 Okay :+1: , But set the default logo to 749x211px , one used in the gif.
screenshot from 2019-01-13 12-42-19

**Comment by YogeshSharma0201** [comment]
@sk9331657 if wanted the logo text can be restricted to some character limit say 10-12, also font size field can be added so that larger logo texts can be used.
But this problem is not just related to logo text:
screenshot from 2019-01-13 14-28-47

Discussion is required on this, about string length limit.

**Comment by rajvaibhavdubey** [comment]
@YogeshSharma0201 NVM, I'll solve this in another PR. This one looks good to go. :+1:

**Pull request merged**